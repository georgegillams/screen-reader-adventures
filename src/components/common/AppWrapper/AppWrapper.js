import React from 'react';
import PropTypes from 'prop-types';

import STYLES from './app-wrapper.scss';

import { cssModules } from 'gg-components/helpers/cssModules';

const getClassName = cssModules(STYLES);

const AppWrapper = props => {
  const { className, ...rest } = props;

  return <div className={getClassName('app__app-wrapper', className)} {...rest} />;
};

AppWrapper.propTypes = {
  className: PropTypes.string,
};

AppWrapper.defaultProps = {
  className: null,
};

export default AppWrapper;
