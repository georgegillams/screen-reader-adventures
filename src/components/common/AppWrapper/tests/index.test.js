import React from 'react';

import { render } from '@testing-library/react';

import AppWrapper from '..';

describe('<AppWrapper />', () => {
  const children = <div>some text</div>;

  it('Should render correctly', () => {
    const { container } = render(<AppWrapper>{children}</AppWrapper>);

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with custom className', () => {
    const { container } = render(<AppWrapper className={'test-custom-classname'}>{children}</AppWrapper>);

    expect(container).toMatchSnapshot();
  });
});
