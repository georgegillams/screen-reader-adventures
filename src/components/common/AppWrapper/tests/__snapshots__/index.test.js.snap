// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppWrapper /> Should render correctly 1`] = `
<div>
  <div
    class="app__app-wrapper"
  >
    <div>
      some text
    </div>
  </div>
</div>
`;

exports[`<AppWrapper /> Should render correctly with custom className 1`] = `
<div>
  <div
    class="app__app-wrapper test-custom-classname"
  >
    <div>
      some text
    </div>
  </div>
</div>
`;
